draft-ietf-sipcore-sip-push.xml

```diff
@@ -438,29 +466,16 @@ align="center"><artwork>
           sending a REGISTER request, the proxy MAY choose to not request a push notification towards the UA (and wait for the associated 
           REGISTER request and 2xx response) before it tries to forward the SIP request towards the UA. The mechanisms for getting such knowledge 
           might be dependent on implementation or deployment architecture, and are outside the scope of this document.
+          Similarly, if the Request-URI of the SIP request only contains any pn-provider SIP URI parameter, but no other pn- SIP URI parameters, e.g., because 
+          the SIP UA has not included them in a REGISTER request (<xref target="section.ua.que"></xref>), the proxy is not able to request a 
+          push notification towards the UA. If the proxy has knowledge that the UA is awake, and that the UA is able to receive the SIP request, 
+          the proxy MAY forwards the request towards the UA. Otherwise the proxy MUST reject the SIP request with a 556 (Push Notification Failed) 
+          response.
         </t>
       </section>
       </section>
     </section>
 
-    <section anchor="section.nat" title="Network Address Translator (NAT) Considerations">
-      <t>
-        Whenever the SIP UA receives a push notification, if the UA is located behind a
-        Network Address Translator (NAT), the UA might need to take actions in order to 
-        establish a binding in the NAT, in order for an incoming SIP request to reach the 
-        UA. By sending the re-registration SIP REQUEST the UA will establish such a NAT binding.
-      </t>
-      <t>
-        If the UA and the SIP proxy are operating in an environment where one or more NATs might
-        be located between the UA and the proxy, when the proxy receives a REGISTER request from 
-        the UA it can store the IP address and port information from where it received the request.
-        Then, when the proxy forwards a SIP request towards the UA, it uses that IP address and port,
-        instead of the IP address and port associated with the Request-URI of the request. If the
-        REGISTER request created one or more NAT bindings, this will ensure that the SIP request reaches
-        the UA.
```

Comment:
I agree that this didn't provide anything but I think removing it was the wrong solution. Aren't we just missing here a couple of sentences about how the proxy may need to reroute based on new NAT pinholes being opened? We could also add a sentence or two on why this solution works nicely with endpoints behind NAT (e.g. a new pinhole will be opened and one doesn't need to be maintained).

I'm not an expert on NAT so it's possible this isn't correct or relevant, but if it is then I think this is worth maintaining.

Reply:
I think Martin's comment was that this is generic behavior when there is NAT, not specific to SIP push.